controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,144 @@
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		// create a noise objects where the labels don't match
+		noiseObject := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match-noise",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		cl := test.NewFakeClient(t, cm, noiseObject)
+
+		// when
+		err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+
+		// then
+		require.NoError(t, err)
+		// check that the configmap doesn't exist anymore
+		AssertObjectNotFound(t, cl, UserSettingNS, "johnsmith", &corev1.ConfigMap{})
+		// check that the noise object still exists
+		retrievedNoise := &corev1.ConfigMap{}
+		AssertObject(t, cl, UserSettingNS, "user-settings-name-no-match-noise", retrievedNoise, func() {
+			assert.Equal(t, noiseObject, retrievedNoise)
+		})
+	})
+	t.Run("multiple objects found by label and deletes successfully", func(t *testing.T) {
+		// given
+		cm1 := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		cm2 := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "user-settings-name-no-match-second",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        "johnsmith",
+				},
+			},
+		}
+		cl := test.NewFakeClient(t, cm1, cm2)
+
+		// when
+		err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+
+		// then
+		require.NoError(t, err)
+		// check that the configmaps don't exist anymore
+		AssertObjectNotFound(t, cl, UserSettingNS, "user-settings-name-no-match", &corev1.ConfigMap{})
+		AssertObjectNotFound(t, cl, UserSettingNS, "user-settings-name-no-match-second", &corev1.ConfigMap{})
+	})
+
+	t.Run("Error is returned when error in deleting configmap", func(t *testing.T) {
+		// given
+		cl := test.NewFakeClient(t)
+		cl.MockGet = func(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+			return nil
+		}
+		cl.MockDelete = func(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
+			return fmt.Errorf("error in deleting configmap")
+		}
+
+		// when
+		err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+
+		// then
+		require.Error(t, err)
+		require.Equal(t, "error in deleting configmap", err.Error())
+	})
+	t.Run("No Error is returned when no object is found", func(t *testing.T) {
+		// given
+		cl := test.NewFakeClient(t)
```

Comment:
Add some noise here too and check it's still there.

Reply:
done in 808948b